[🐛 Bug]: WDIO 9.4.3 and up no longer picks wdio.conf.ts config file by default, tries to use wdio.conf.js instead

### Have you read the Contributing Guidelines on issues?

- [x] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

9.4.3

### Node.js Version

22.13.1

### Mode

WDIO Testrunner

### Which capabilities are you using?

```typescript

```

### What happened?

When I migrated to using WDIO 9 from WDIO 8 using Typescript, I was happy to learn that I didn't need to specify the config file `wdio.conf.ts` file on the command line each time I ran WDIO. This was true with `@wdio-cli` 9.4.2.

Starting with 9.4.3, this got broken and now the default configuration file that it is looking for is `wdio.conf.js`, which I think is backward steps since the project itself and most people are trying to move to using TypeScript instead of Javascript. The shortened error message is this:

 ```bash
ERROR @wdio/config:ConfigParser: Failed loading configuration file: file:///Users/ianrenauld/dev/wdio9/wdio.conf.js: Cannot find module '/Users/ianrenauld/dev/wdio9/wdio.conf.js' imported from /Users/ianrenauld/dev/wdio9/node_modules/@wdio/config/build/node/index.js
```


### What is your expected behavior?

Using this command-line:

```bash
npx wdio --spec tests/simple.spec.ts
```

will start the test properly

### How to reproduce the bug.

On a WDIO 9 project (you can get my minimal project to reproduce easily here: https://github.com/ianrenauld/wdio9, but this will happen on any WDIO 9 repo using 9.4.3 and up) do:

```bash
npx wdio --spec tests/simple.spec.ts
```

Workaround to the bug is to specify the config file each time on the CLI:

```bash
npx wdio wdio.conf.ts --spec tests/simple.spec.ts
```

To see previous (working behavior), change `package.json` and set the `wdio-cli` line to this:

```bash
    "@wdio/cli": "9.4.2",
```

Do `npm install` and try running again without specifying the config file, it will work.


### Relevant log output

```typescript
2025-02-11T21:52:38.093Z ERROR @wdio/config:ConfigParser: Failed loading configuration file: file:///Users/ianrenauld/dev/wdio9/wdio.conf.js: Cannot find module '/Users/ianrenauld/dev/wdio9/wdio.conf.js' imported from /Users/ianrenauld/dev/wdio9/node_modules/@wdio/config/build/node/index.js
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/ianrenauld/dev/wdio9/wdio.conf.js' imported from /Users/ianrenauld/dev/wdio9/node_modules/@wdio/config/build/node/index.js
    at finalizeResolution (node:internal/modules/esm/resolve:275:11)
    at moduleResolve (node:internal/modules/esm/resolve:932:10)
    at defaultResolve (node:internal/modules/esm/resolve:1056:11)
    at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:654:12)
    at #cachedDefaultResolve (node:internal/modules/esm/loader:603:25)
    at ModuleLoader.resolve (node:internal/modules/esm/loader:586:38)
    at ModuleLoader.getModuleJobForImport (node:internal/modules/esm/loader:242:38)
    at onImport.tracePromise.__proto__ (node:internal/modules/esm/loader:546:36)
    at TracingChannel.tracePromise (node:diagnostics_channel:344:14)
    at ModuleLoader.import (node:internal/modules/esm/loader:545:21) {
  code: 'ERR_MODULE_NOT_FOUND',
  url: 'file:///Users/ianrenauld/dev/wdio9/wdio.conf.js'
}
```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [x] I have searched the existing issues

--- a/packages/wdio-cli/src/commands/config.ts
+++ b/packages/wdio-cli/src/commands/config.ts
@@ -214,7 +214,6 @@ export async function canAccessConfigPath(configPath: string) {
  * @param {Function} runConfigCmd   runConfig method to be replaceable for unit testing
  */
 export async function missingConfigurationPrompt(command: string, configPath: string, runConfigCmd = runConfigCommand) {
-
     const message = (
         `Could not execute "${command}" due to missing configuration, file ` +
         `"${path.parse(configPath).name}[.js/.ts]" not found! ` +
@@ -236,9 +235,13 @@ export async function missingConfigurationPrompt(command: string, configPath: st
         console.log(`No WebdriverIO configuration found in "${process.cwd()}"`)
 
         /* istanbul ignore next */
-        return !process.env.WDIO_UNIT_TESTS && process.exit(0)
+        if (!process.env.WDIO_UNIT_TESTS) {
+            process.exit(0)
+        }
+        return configPath
     }
 
     const parsedAnswers = await parseAnswers(false)
     await runConfigCmd(parsedAnswers, 'latest')
+    return configPath
 }

--- a/packages/wdio-cli/src/commands/run.ts
+++ b/packages/wdio-cli/src/commands/run.ts
@@ -180,6 +180,11 @@ export async function handler(argv: RunCommandArguments) {
     if (!confAccess) {
         try {
             await missingConfigurationPrompt('run', wdioConf.fullPathNoExtension)
+            if (process.env.WDIO_UNIT_TESTS) {
+                return
+            }
+
+            return handler(argv)
         } catch {
             process.exit(1)
         }
@@ -196,7 +201,7 @@ export async function handler(argv: RunCommandArguments) {
         process.env.TSX_TSCONFIG_PATH && path.resolve(process.cwd(), process.env.TSX_TSCONFIG_PATH)
     )
     const tsConfigPathFromParams = params.tsConfigPath && path.resolve(process.cwd(), params.tsConfigPath)
-    const tsConfigPathRelativeToWdioConfig = path.join(path.dirname(wdioConf.fullPath), 'tsconfig.json')
+    const tsConfigPathRelativeToWdioConfig = path.join(path.dirname(confAccess), 'tsconfig.json')
     const localTSConfigPath = (
         tsConfigPathFromEnvVar ||
         tsConfigPathFromParams ||
@@ -211,7 +216,7 @@ export async function handler(argv: RunCommandArguments) {
      * if `--watch` param is set, run launcher in watch mode
      */
     if (params.watch) {
-        const watcher = new Watcher(wdioConf.fullPath, params)
+        const watcher = new Watcher(confAccess, params)
         return watcher.watch()
     }
 
@@ -223,12 +228,12 @@ export async function handler(argv: RunCommandArguments) {
      * stdin.isTTY is false when command is from nodes spawn since it's treated as a pipe
      */
     if (process.stdin.isTTY || !process.stdout.isTTY) {
-        return launch(wdioConf.fullPath, params)
+        return launch(confAccess, params)
     }
 
     /*
      * get a list of spec files to run from stdin, overriding any other
      * configuration suite or specs.
      */
-    launchWithStdin(wdioConf.fullPath, params)
+    launchWithStdin(confAccess, params)
 }

--- a/packages/wdio-cli/tests/commands/run.test.ts
+++ b/packages/wdio-cli/tests/commands/run.test.ts
@@ -65,6 +65,12 @@ describe('Command: run', () => {
             .toContain('sample.conf')
     })
 
+    it('should check for js and ts default config files', async () => {
+        vi.mocked(fs.access).mockRejectedValueOnce('not found')
+        const result = await runCmd.handler({ configPath: 'sample.conf.js' } as any)
+        expect(result).toContain('sample.conf.ts')
+    })
+
     it('should use local conf if nothing defined', async () => {
         const launcher = await runCmd.handler({ argv: {} } as any)
         expect(launcher).toContain('wdio.conf.js')

